ui/src/admin/containers/AdminChronografPage.js

```diff
@@ -0,0 +1,383 @@
+      organization: this.props.currentOrganization,
+      selectedUsers: [],
+      filteredUsers: this.props.users,
+      showManageOverlay: false,
+      showCreateUserOverlay: false,
+    }
+  }
+
+  // TODO: revisit this, possibly don't call setState if both are deep equal
+  componentWillReceiveProps(nextProps) {
+    const {users, currentOrganization} = nextProps
+
+    this.handleFilterUsers({
+      users,
+      organization: currentOrganization,
+    })
+  }
+
+  componentDidMount() {
+    const {
+      links,
+      actions: {loadUsersAsync, loadOrganizationsAsync},
+    } = this.props
+
+    loadUsersAsync(links.users)
+    loadOrganizationsAsync(links.organizations) // TODO: make sure server allows admin to hit this for safety
+  }
+
+  handleShowManageOrgsOverlay = () => {
+    this.setState({showManageOverlay: true})
+  }
+  handleShowCreateUserOverlay = () => {
+    this.setState({showCreateUserOverlay: true})
+  }
+  handleHideOverlays = () => {
+    this.setState({showManageOverlay: false, showCreateUserOverlay: false})
+  }
+
+  handleFilterUsers = ({users, organization}) => {
+    const nextUsers = users || this.props.users
+    const nextOrganization = organization || this.props.currentOrganization
+
+    const filteredUsers =
+      nextOrganization.name === DEFAULT_ORG_NAME
+        ? nextUsers
+        : nextUsers.filter(
+            user =>
+              nextOrganization.name === NO_ORG
+                ? user.roles.length === 1 // Filter out if user is only part of Default Org
+                : user.roles.find(
+                    role => role.organization === nextOrganization.id
+                  )
+          )
+    this.setState({
+      filteredUsers,
+      organization: nextOrganization,
+      selectedUsers:
+        nextOrganization.name === DEFAULT_ORG_NAME
+          ? this.state.selectedUsers
+          : [],
+    })
+  }
+
+  handleToggleUserSelected = user => e => {
+    e.preventDefault()
+
+    const {selectedUsers} = this.state
+
+    const isUserSelected = selectedUsers.find(u => isSameUser(user, u))
+
+    const newSelectedUsers = isUserSelected
+      ? selectedUsers.filter(u => !isSameUser(user, u))
+      : [...selectedUsers, user]
+
+    this.setState({selectedUsers: newSelectedUsers})
+  }
+  handleToggleAllUsersSelected = areAllSelected => () => {
+    const {filteredUsers} = this.state
+
+    if (areAllSelected) {
+      this.setState({selectedUsers: []})
+    } else {
+      this.setState({selectedUsers: filteredUsers})
+    }
+  }
+
+  // SINGLE USER ACTIONS
+  handleCreateUser = user => {
+    const {links, actions: {createUserAsync}} = this.props
+    let newUser = user
+
+    if (
+      user.roles.length === 1 &&
+      user.roles[0].organization !== DEFAULT_ORG_ID
+    ) {
+      newUser = {
+        ...newUser,
+        roles: [
+          ...newUser.roles,
+          {organization: DEFAULT_ORG_ID, name: MEMBER_ROLE},
```

Comment:
You may want to add a comment explaining why you're doing this

Reply:
removed that logic as it's for an obsolete approach